core/arns/src/main/java/software/amazon/awssdk/arns/ArnResource.java

```diff
@@ -91,34 +91,30 @@ public static Builder builder() {
      * @return {@link ArnResource}
      */
     public static ArnResource fromString(String resource) {
-        Character splitter = StringUtils.findFirstOccurrence(resource, ':', '/');
+        Character primarySplitter = StringUtils.findFirstOccurrence(resource, ':', '/');
 
-        if (splitter == null) {
-            return ArnResource.builder().resource(resource).build();
+        if (primarySplitter == null) {
+            return builder().resource(resource).build();
         }
 
-        int resourceTypeColonIndex = resource.indexOf(splitter);
+        int resourceTypeIndex = resource.indexOf(primarySplitter);
+        ArnResource.Builder builder = builder().resourceType(resource.substring(0, resourceTypeIndex));
 
-        ArnResource.Builder builder = ArnResource.builder().resourceType(resource.substring(0, resourceTypeColonIndex));
-        int resourceColonIndex = resource.indexOf(splitter, resourceTypeColonIndex);
-        int qualifierColonIndex = resource.indexOf(splitter, resourceColonIndex + 1);
-        if (qualifierColonIndex < 0) {
-            builder.resource(resource.substring(resourceTypeColonIndex + 1));
+        String remainingResource = resource.substring(resourceTypeIndex + 1);
+        int qualifierColonIndex = remainingResource.lastIndexOf(':');
+        if (qualifierColonIndex < 1) {
+            builder.resource(remainingResource);
         } else {
-            builder.resource(resource.substring(resourceTypeColonIndex + 1, qualifierColonIndex));
-            builder.qualifier(resource.substring(qualifierColonIndex + 1));
+            builder.resource(remainingResource.substring(0, qualifierColonIndex));
+            builder.qualifier(remainingResource.substring(qualifierColonIndex + 1));
         }
 
         return builder.build();
     }
 
     @Override
     public String toString() {
-        return this.resourceType
-               + ":"
-               + this.resource
-               + ":"
-               + this.qualifier;
+        return String.join(":", this.resourceType, this.resource, this.qualifier);
```

Comment:
Could this potentially break customers? Should we introduce a new method, say, toStringPretty or toStringFormatted?

Reply:
Why would it break people? The output looks like it would be the same to me. I don't feel strongly about it though, it was a suggestion from IntelliJ.

Reply:
Oh nvm, I thought we made changes to format the arn for some reason. That must be from a separate PR...